### ISSUE TITLE
Tridactyl does not recognise custom search engines in Firefox

### ISSUE THREAD

**Issue by Noexpert** [tridactyl/tridactyl#4652]
Ver: 1.23.0

I have some custom search engines in Firefox with assigned keywords.
They are automatically recognised in Tridactyl **without** setting them again in Tridactyl.
I don't remember how I added them, but I believe I used the "old" method. before some changes were made to Firefox.

I have now added a new search engine using the method here:
https://superuser.com/a/1756774

The search url is:
`https://www.ebay.co.uk/sch/i.html?_nkw=%s&_sacat=0&_sop=15&_blrs=recall_filtering&_dmd=2&rt=nc&LH_PrefLoc=1`

and the keyword is `eb`.

This works correctly in Firefox, but the `eb` keyword is not recognised by Tridactyl, even though the keywords for other custom engines are.

When I try `eb test` it takes me to `https://www.google.co.uk/search?q=eb+test`, instead of recognising the `eb` keyword.

**Comment by bovine3dom** [comment]
On the new tab page could you open the web console and run `tri.l(tri.browserBg.search.get())` and see if your search engine appears?

If it doesn't show up there's not much we can do, either your method actually adds the unsupported (Mozilla's choice, not ours) bookmark search engines or there is an upstream bug that we can report.

**Comment by Noexpert** [comment]
Did you mean in the Firefox web console?
If so, I'm getting this error: `Uncaught ReferenceError: tri is not defined`

**Comment by bovine3dom** [comment]
I do mean the web console, are you sure you're on our new tab page? If you're using our no-new-tab build you can do it on the `:help` page instead.

The method I suggested won't work on a normal page.

On any page you could run `:js tri.l(tri.browserBg.search.get())` in our command line and then look in the web console for the result.

**Comment by Noexpert** [comment]
My mistake; I was not on the new tab page.

Here's an extract of the output:
```
Object { name: "google.co.uk", isDefault: true, alias: "g", … }
Object { name: "Wikipedia (en)", isDefault: false, alias: "w", … }
Object { name: "eBay UK", isDefault: false, alias: undefined, … }
```

The alias is `undefined`, but note that it does work in the Firefox address bar.
Unclear why.

**Comment by bovine3dom** [comment]
If you go to `about:preferences#search`, does it have a keyword assigned? you should be able to set one by double clicking in the blank area if there isn't one

**Comment by Noexpert** [comment]
Yes, the keyword is `eb`.
This works correctly in the Firefox address bar, but not in Tridactyl.

**Comment by bovine3dom** [comment]
I've looked into this a bit and there are two/three problems:

1) Tridactyl makes no attempt to search using anything other than the default search engine (we expect you to use `searchurls`, see `:help searchurls`).

2) `browser.search.get()` returns `undefined` aliases for default aliases and for manually added engines using your flag

3) search engines can have multiple aliases but `search.get()` can only return one per engine

The about:config flag you're using was added in https://bugzilla.mozilla.org/show_bug.cgi?id=1106626 and is a decent jumping off point for related issues.

We should hunt around to see if there are any issues already open about this stuff on bugzilla and file our own if not.

I wouldn't get too hopeful about it being fixed any time soon as I think Mozilla have been talking about overhauling the search engine stuff for over a decade now :)

Related: #760
